ragna/core/_rag.py

```diff
@@ -188,7 +199,17 @@ async def prepare(self) -> Message:
                 detail=RagnaException.EVENT,
             )
 
-        await self._run(self.source_storage.store, self.documents)
+        from ragna.core import Document
+        if issubclass(self.source_storage.__ragna_input_type__, Document):
+            await self._run(self.source_storage.store, self.documents)
+        else:
+            # Here we need to generate the list of embeddings
+            chunks = []
+            for document in self.documents:
+                chunks += self.embedding_model._chunk_pages(document.extract_pages(), document_id=document.id, chunk_size=500, chunk_overlap=250)
+            embeddings = self.embedding_model.embed_chunks(chunks)
```

Comment:
I've commented this before: this logic cannot be here. It should be hardcoded in the embed method of MiniLML6v2. Of course will not stay there when we do the chunking refactor. But for now, we just want to preserve the status quo. `self.embedding_model._chunk_pages` is not something we can depend on here given that it is not part of the standard API of the `EmbeddingModel`

Reply:
I see what you mean, I misunderstood.  I will fix it.